app/api/admin_proxies.py

```
@@ -0,0 +1,530 @@
+            ),
+            "avg_response_time_ms": round(stats.avg_response_time or 0, 2),
+            "avg_success_rate": round(stats.avg_success_rate or 0, 2)
+        },
+        "country_distribution": country_distribution,
+        "provider_performance": provider_performance,
+        "insights": insights
+    }
+
+@router.post("/rotate")
+async def rotate_proxy_assignments(
+    scraper_ids: List[str] = Query(description="Specific scrapers to rotate proxies for"),
```

Ruff (0.13.2)

Do not perform function call `Query` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)